nixtla/nixtla_client.py

```diff
@@ -994,7 +1035,22 @@ def forecast(
             )
             in_sample_df = ufp.drop_columns(in_sample_df, target_col)
             out = ufp.vertical_concat([in_sample_df, out])
-            out = ufp.sort(out, by=[id_col, time_col])
+        self._maybe_assign_feature_contributions(
+            feature_contributions=resp["feature_contributions"],
+            x_cols=x_cols,
+            out_df=out[[id_col, time_col, "TimeGPT"]],
+            insample_feat_contributions=insample_feat_contributions,
+        )
+        if add_history:
+            sort_idxs = ufp.maybe_compute_sort_indices(
+                out, id_col=id_col, time_col=time_col
+            )
+            assert sort_idxs is not None
+            out = ufp.take_rows(out, sort_idxs)
+            self.feature_contributions = ufp.take_rows(
+                self.feature_contributions, sort_idxs
+            )
```

Comment:
sorry, I messed up the suggestion
```suggestion
            if sort_idxs is not None:  # this is None for a single serie
                out = ufp.take_rows(out, sort_idxs)
                if feature_contributions:
                    self.feature_contributions = ufp.take_rows(
                        self.feature_contributions, sort_idxs
                    )
```